src/Mod/Sketcher/Gui/DrawSketchHandler.cpp

```diff
@@ -420,139 +420,125 @@ DrawSketchHandler::suggestedConstraintsPixmaps(std::vector<AutoConstraint>& sugg
                 constr.Type = lineCenter ? Sketcher::Symmetric : Sketcher::PointOnObject;
             }
             else {
                 constr.Type = Sketcher::Coincident;
             }
         }
-        else if (type == AutoConstraint::CURVE && PosId != PointPos::none) {
+        else if (type == AutoConstraint::CURVE && preSel.posId != PointPos::none) {
             constr.Type = Sketcher::PointOnObject;
         }
-        else if (type == AutoConstraint::CURVE && PosId == PointPos::none) {
+        else if (type == AutoConstraint::CURVE && preSel.posId == PointPos::none) {
             constr.Type = Sketcher::Tangent;
         }
 
-        if (constr.Type == Sketcher::Tangent && preselectIsLine) {
-            if (Dir.Length() < 1e-8 || hitShapeDir.Length() < 1e-8) {
-                // Direction not set so return;
-                return suggestedConstraints.size();
+        if (constr.Type == Sketcher::Tangent && preSel.isLine) {
+            if (Dir.Length() < 1e-8 || preSel.hitShapeDir.Length() < 1e-8) {
+                return;  // Direction not set so return;
             }
+
             // We are hitting a line and have hitting vector information
             Base::Vector3d dir3d = Base::Vector3d(Dir.x, Dir.y, 0);
-            double cosangle = dir3d.Normalize() * hitShapeDir.Normalize();
-
-            // the angle between the line and the hitting direction are over around 6 degrees (it is
-            // substantially parallel) or if it is an sketch axis (that can not move to accommodate
-            // to the shape), then only if it is around 6 degrees with the normal (around 84
-            // degrees)
-            if (fabs(cosangle) < 0.995f
-                || ((GeoId == Sketcher::GeoEnum::HAxis || GeoId == Sketcher::GeoEnum::VAxis)
-                    && fabs(cosangle) < 0.1)) {
-                suggestedConstraints.push_back(constr);
-            }
-
+            double cosangle = dir3d.Normalize() * preSel.hitShapeDir.Normalize();
 
-            return suggestedConstraints.size();
+            // the angle between the line and the hitting direction are over around 6 degrees
+            if (fabs(cosangle) > 0.995f) {
+                return;
+            }
         }
 
         if (constr.Type != Sketcher::None) {
             suggestedConstraints.push_back(constr);
         }
     }
+}
 
-    if (Dir.Length() < 1e-8 || type == AutoConstraint::CURVE) {
-        // Direction not set so return;
-        return suggestedConstraints.size();
-    }
-
-    // Suggest vertical and horizontal constraints
-
-    // Number of Degree of deviation from horizontal or vertical lines
-    const double angleDev = 2;
-    const double angleDevRad = angleDev * pi / 180.;
+void DrawSketchHandler::seekAlignmentAutoConstraint(
+    std::vector<AutoConstraint>& suggestedConstraints,
+    const Base::Vector2d& Dir)
+{
+    using std::numbers::pi;
+    const double angleDevRad = 0.035;  // 2 degrees in radians

```

Comment:
Maybe best to just keep it to the old value (i.e. converted from degrees to radians). If not, at least remove `using std::numbers::pi`.

Reply:
This removes useless computation from a function that is run at mousemove. pi is used in other places of the function.

Reply:
I wouldn't call it useless computation, not without providing the full 16 digits past decimal for the result. However for UI purposes it will hopefully not matter. So I won't insist further.

OK on `pi`.

Reply:
2deg was choosen arbitrarily. It is not better in anyway than 2.0053523.

Reply:
My point was to avoid the computation that is not necessary since 2.0 is a magic number. Since we are using a magic number, might as well skip the computation that occurs on every mouse move for nothing.